### ISSUE TITLE
[Enhancement Request] Bulk move support from $source_volume to $target_volume

### ISSUE THREAD

**Issue by DanBurkhardt** [007revad/Synology_app_mover#83]
Reading the docs, it seems like you've done some dependency mapping to ensure things are re-linked properly (haven't looked at the scripts though, just assuming).

In any case, since this tool has methods in place to move applications safely and re-link deps, I think it would be a great enhancement to be able to bulk-move applications, where all applications from /$source_volume are moved to /$target_volume in one go.

ex:

```shell
sudo -s ./syno_app_mover.sh --bulk-move --source /volume5 -- target /volume3
```

**Comment by 007revad** [comment]
A lot of work has gone into this script to make sure that extra folders, symlinks and settings are moved and edited to suit the new volume location.

I added an "all" menu option for backing up and restoring. Adding an 'all' menu option for moving would be easy enough to add. You're the 2nd person in 2 weeks that have asked for it.

**Comment by DanBurkhardt** [comment]
Yeah clearly a lot has! I started with an app I didn't care about getting hosed and it worked perfectly, scaled up from that and eventually moved everything. 

To confirm-- we're good to delete the volume after the move right? No data tendrils should be left (understanding some apps need you to move the db, like drive sync)

**Comment by 007revad** [comment]
I recommend checking your apps still work as expected, without the old volume, before deleting the old volume: 
1. Shut down the NAS.
2. Remove the drives that contain the old volume.
3. Boot the NAS.
4. Mute the beep.
5. Check none of your packages show Broken or Repair in storage manager.
6. Check each app still has it's settings and data.

Assuming all your apps are okay: 
1. Shut down the NAS.
2. Reinstall the drives that contain the old volume.
3. Boot the NAS.
4. Delete the volume.

Of course all of the above assumes that there is only 1 volume on those drives.

**Comment by presidento** [comment]
I'm the third one. I upgraded from Synology DS220j (2 bay) to DS923+ (4 bay) and want to migrate everything from ext4 to btrfs, so I created a new btrfs volume with 2 new drives and I have to move everything there from the old ext4 volume. I'll do it one by one now. Beside your project there is only one solution: backup, reinstall/reconfigure, and restore everything, which not optimal if I have 2 working disk and 2 new disk in a 4 bay NAS to make the transition. Thank you for your work.

**Comment by gerstemann** [comment]
+1 for moving all packages from one to another volume.

**Comment by joelhinch** [comment]
+1 also please when you have time this would be fantastic

**Comment by marcelrv** [comment]
Another +1 for a move all

**Comment by filol** [comment]
+1

**Comment by herzkerl** [comment]
+1

**Comment by FoldZero** [comment]
+1
My situation is 20 apps, all on Volume 2, I would like a switch to say, Automove Vol2 > Vol1.

**Comment by ChaosEmpire** [comment]
I forgot to switch my discs to 4k, so i moved all data to another nas
left 10gb, and wanted to do a nas backup
it took a day and still not finished

i decided to degrade raid 6 to "raid 5"  and converted 1 disk, got the failure, reinstalled
now i use your script to move all packages...sadly 1 by 1 to the "new 4k" disk
and even i have to do all manually 1 by 1 its so much faster to get all data moved without further unknown hours, and tb of saved data..from a voume with just 10gb left 

Please implement the move all ...the next one would not need to click every app by hand over and over ;)

Thanks a lot for your much appreciated work
